v3-docs/docs/tutorials/meteorjs-vue.md

```diff
@@ -0,0 +1,1375 @@
+
+const addTask = async () => {
+  await Meteor.callAsync('insertTask', newTask.value.trim());
+  newTask.value = '';
+}
+</script>
+```
+:::
+
+Inside the function, we are adding a task to the `tasks` collection by calling `Meteor.callAsync()`. The first argument is the name of the method we want to call, and the second argument is the text of the task. We are also trimming the text to remove any extra spaces.
+
+### 3.5: Show Newest Tasks First
+
+
+Now, you just need to make a change which will improve user experience: we will show the newest tasks first. We can accomplish this quickly by sorting our [MongoDB](https://guide.meteor.com/collections.html#mongo-collections) query.
+
+
+::: code-group
+```javascript [imports/ui/App.vue]
+...
+
+const tasks = autorun(() => {
+  return TasksCollection.find({}, { sort: { createdAt: -1 } }).fetch();
+}).result;
+
+...
+```
+:::
+
+Your app should look like this:
+
+![image](https://vue3-tutorial.meteor.com/simple-todos/assets/new-screenshots/step03/newest-task.png)
+
+## 4: Update and Remove
+
+Until now, you have only inserted documents to our collection. Let’s see how you can update and remove them using the user interface.
+
+
+### 4.1: Add Checkbox
+
+
+First, you need to add a `checkbox` element to your `Task` component, and we need to add the `v-model` directive to the checkbox. This will allow us to bind the value of the checkbox to the `checked` field of the task document.  
+To do this, we need to add a `ref` to the task document. This will allow us to access the task document in the template. And add a computed property `isChecked` for the state management of the checkbox.
+
+We also have a prop called `task` that is passed to the component. This prop is an object that represents the task document.
+
+
+::: code-group
+```vue [imports/ui/components/Task.vue]
+<script setup>
+import { ref, computed } from 'vue';
+
+const props = defineProps({
+  task: {
+    type: Object,
+    required: true,
+  },
+});
+
+const taskRef = ref({ ...props.task }); // Create a reactive copy of the task
+
+const isChecked = computed(() => taskRef.value.checked);
+
+</script>
+
+<template>
+  <div class="flex items-center rounded px-4 py-2 mb-2">
+    <li>
+      <input v-model="taskRef.checked" type="checkbox" :checked="isChecked" @change="handleCheckboxChange" />
+    </li>
+    <span class="text-gray-600 pl-2" :class="{ 'text-gray-400 italic line-through': taskRef.checked }">
+      {{ task.text }}
+    </span>
+  </div>
+</template>
+ ..
+```
+:::
+
+### 4.2: Toggle Checkbox
+
+
+We need to implement the method to update the task document. So, update the `tasksMethods.js` file with the following code:
+
+
+::: code-group
+```javascript [imports/api/tasksMethods.js]
+import { Meteor } from 'meteor/meteor';
+import { check } from 'meteor/check';
+import { TasksCollection } from './tasksCollection';
+
+async function insertTask(text) {
+    check(text, String);
+    return await TasksCollection.insertAsync({
```

Comment:
What do you think about to put await explicitely on returns? Usually linters suggest you even remove them when using async, that is implicitely understood, but I don't know what you think.

Reply:
Including `await` can make it immediately clear to anyone reading the code that the function is performing an async operation and that the return value is a promise. I think this can improve readability, especially for developers who may not be as familiar with the codebase.